src/sync.rs

```diff
@@ -557,14 +565,30 @@ impl<T> OnceLockExt<T> for std::sync::OnceLock<T> {
     }
 }
 
+impl<T> MutexExt<T> for std::sync::Mutex<T> {
+    fn lock_py_attached(
+        &self,
+        _py: Python<'_>,
+    ) -> std::sync::LockResult<std::sync::MutexGuard<'_, T>> {
+        // SAFETY: detach from the runtime right before a possibly blocking call
```

Comment:
Instead of unconditionally detaching, is it worth calling `self.try_lock()` first, and returning eagerly in all cases except where it would block? (And then if it would block, continue to the code below.)

Reply:
I guess it can't hurt to avoid detaching and reattaching if it's possible. I'll go ahead and do that.